sql/table.go

```diff
@@ -164,6 +164,8 @@ func encodeTableKey(b []byte, v parser.Datum) ([]byte, error) {
 		return encoding.EncodeNumericFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeBytes(b, []byte(t)), nil
+	case parser.DNull:
+		return b, nil
```

Comment:
so actually we have a bigger problem - we can't just do nothing. We need to encode this nil as a "typed" nil, and for that we need to pass the type around or else figure out how to return something typed from `Arg`. Basically, we need access to the descriptor to figure out how to interpret the nil.